src/sentry/dynamic_sampling/tasks/boost_low_volume_projects.py

```diff
@@ -234,92 +242,17 @@ def fetch_projects_with_total_root_transaction_count_and_rates(
-        else:
-            raise ValueError(f"Unsupported measure: {measure}")
-
-        where = [
-            Condition(Column("timestamp"), Op.GTE, datetime.utcnow() - query_interval),
-            Condition(Column("timestamp"), Op.LT, datetime.utcnow()),
-            Condition(Column("metric_id"), Op.EQ, metric_id),
-            Condition(Column("org_id"), Op.IN, org_ids),
-        ]
-
-        keep_count = Function(
-            "sumIf",
-            [
-                Column("value"),
-                Function("equals", [Column(transaction_tag), "keep"]),
-            ],
-            alias="keep_count",
-        )
-        drop_count = Function(
-            "sumIf",
-            [
-                Column("value"),
-                Function("equals", [Column(transaction_tag), "drop"]),
-            ],
-            alias="drop_count",
-        )
-
-        while time.monotonic() < context.expiration_time:
-            query = (
-                Query(
-                    match=Entity(EntityKey.GenericOrgMetricsCounters.value),
-                    select=[
-                        Function("sum", [Column("value")], "root_count_value"),
-                        Column("org_id"),
-                        Column("project_id"),
-                        keep_count,
-                        drop_count,
-                    ],
-                    groupby=[Column("org_id"), Column("project_id")],
-                    where=where,
-                    granularity=granularity,
-                    orderby=[
-                        OrderBy(Column("org_id"), Direction.ASC),
-                        OrderBy(Column("project_id"), Direction.ASC),
-                    ],
-                )
-                .set_limitby(
-                    LimitBy(
-                        columns=[Column("org_id"), Column("project_id")],
-                        count=MAX_TRANSACTIONS_PER_PROJECT,
-                    )
-                )
-                .set_limit(CHUNK_SIZE + 1)
-                .set_offset(offset)
-            )
-            request = Request(
-                dataset=Dataset.PerformanceMetrics.value,
-                app_id="dynamic_sampling",
-                query=query,
-                tenant_ids={"use_case_id": UseCaseID.TRANSACTIONS.value, "cross_org_query": 1},
-            )
-            data = raw_snql_query(
-                request,
-                referrer=Referrer.DYNAMIC_SAMPLING_DISTRIBUTION_FETCH_PROJECTS_WITH_COUNT_PER_ROOT.value,
-            )["data"]
-            count = len(data)
-            more_results = count > CHUNK_SIZE
-            offset += CHUNK_SIZE
-
-            if more_results:
-                data = data[:-1]
-
-            for row in data:
+        for chunk in TimedIterator(
+            context,
+            query_projects_with_total_root_transaction_count_and_rates(
+                org_ids,
+                measure,
+                query_interval,
+            ),
+            func_name,
+        ):
```

Comment:
_nit_: This could be more readable if written like so:

```suggestion
        iter = query_projects_with_total_root_transaction_count_and_rates(org_ids, measure, query_interval)
        for chunk in TimedIterator(context, iter, func_name):
```